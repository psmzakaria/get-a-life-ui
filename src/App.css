body {
  margin: 0;
  padding: 0;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
}

.text-align-center {
  text-align: center;
}

.cursor-pointer {
  cursor: pointer;
}

.eorganiser-container {
  padding: 20px;
}

.esnap-container-size {
  width: 200px;
  height: 100%;
  margin: 16px;
  cursor: pointer;
}

.esnap-create-event {
  width: 200px;
  height: 100%;
  margin: auto;
  cursor: pointer;
}

.esnap-icon {
  margin: auto;
  margin-top: 20px;
  margin-bottom: 20px
}

.profile-container {
  height: 100vh;
}

.container {
  margin: 0;
  padding: 160px;
  background-image: url("adventure-balloon-clouds-68806\ \(1\).jpg");
  background-size: cover;
  /* border: 2px solid red;
  border-radius: 10px; */
  display: grid;
}

.sideInfo {
  height: 75vh;
  padding: 5px;
  text-align: center;
  /* display: grid;
  grid-template-rows: 1fr 2fr; */
}

.sideinfofirst {
  color: white;
  padding-right: 70px;
}

.sideinfomiddle {
  color: white;
  padding-left: 60px;
}

.sideinfolast {
  color: white;
  padding-right: 40px;
}

.logosize {
  width: 70px;
}

@media only screen and (max-width: 767px) {
  .container {
    margin: 25px;
  }
  .profile-container {
    margin-left: auto;
    margin-right: auto;
    height: 100%;
  }
}

 {
  /* background-color: rgba(69, 42, 165, 0.384); */
}

/* .App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
} */