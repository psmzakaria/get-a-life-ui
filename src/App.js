import React, { Component } from "react";
import { API_URL } from "./utils/configVar";
import UserSignup from "./components/UserSignup";
import "./App.css";
import CreateEvent from "../src/components/CreateEvent"

class App extends Component {
  constructor() {
    super();
    this.state = {
      message: ""
    };
  }

  async componentDidMount() {
    const response = await fetch(API_URL);
    if (response.ok) {
      const data = await response.json();

      this.setState({
        message: data.message
      });
    }
  }

  render() {
    return (
      <div className="App">
        <p className="App-intro">{this.state.message}</p>
        <CreateEvent />
        <div className="App-intro">{this.state.message}</div>
        <UserSignup/>
      </div>
    );
  }
}
it("renders title and authorlist after the component mount", async () => {
  fetch.mockResponseOnce(
    JSON.stringify([
      {
        name: "roald Dhal",
        age: 29,
        _id:"47389577"
      },
      {
        name: "Dolly Lama",
        age: 99,
        _id:"473895789547"
      }
    ])
  );

  const renderer = new ShallowRenderer();
  renderer.render(<Author />);
  const instance = renderer.getMountedInstance();
  await instance.componentDidMount();
  const output = renderer.getRenderOutput();
  expect(output).toMatchSnapshot();
});

export default App;
