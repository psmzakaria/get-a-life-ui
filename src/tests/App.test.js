import React from "react";
import App from "../App";
import ShallowRenderer from "react-test-renderer/shallow";

it("renders without crashing", async () => {
  fetch.mockResponseOnce(
    JSON.stringify({
      message: "This is a test"
    })
  );

  const renderer = new ShallowRenderer();
  renderer.render(<App />);

  const instance = renderer.getMountedInstance();
  await instance.componentDidMount();

  const output = renderer.getRenderOutput();

  expect(output).toMatchSnapshot();
});
