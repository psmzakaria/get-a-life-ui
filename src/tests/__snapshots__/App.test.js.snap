// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="App"
>
  <div
    className="App-intro"
  >
    This is a test
  </p>
  <CreateEvent />
  </div>
  <UserSignup />
</div>
`;
